shell/common/layer_tree_holder.cc

```diff
@@ -0,0 +1,28 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/common/layer_tree_holder.h"
+
+namespace flutter {
+
+std::unique_ptr<LayerTree> LayerTreeHolder::Get() {
+    std::scoped_lock lock(layer_tree_mutex);
+    return std::move(layer_tree_);
+}
+
+void LayerTreeHolder::Replace(std::unique_ptr<LayerTree> layer_tree) {
+    std::scoped_lock lock(layer_tree_mutex);
```

Comment:
Perhaps check the target times of the existing and new layer_trees and only replace if newer (and then rename the method accordingly)?

Reply:
good idea.